### PR TITLE
refactor: Migrate to test suites

### LINKED ISSUE #48: Fix use of deprecated gradle features
```
> Task :kgraphql:test
The automatic loading of test framework implementation dependencies has been deprecated. This is scheduled to be removed in Gradle 9.0. Declare the desired test framework directly on the test suite or explicitly declare the test framework implementation dependencies on the test's runtime classpath. Consult the upgrading guide for further information: https://docs.gradle.org/8.10.2/userguide/upgrading_version_8.html#test_framework_implementation_dependencies
```

### PR DESCRIPTION
**Pull request by stuebingerb**
Migrate to test suites to get rid of deprecation warnings and as Gradle's way forward of testing.

Resolves #48

### CODE REVIEW

File: buildSrc/src/main/kotlin/kotlin-conventions.gradle.kts [hunk 1]
```diff
@@ -1,22 +1,19 @@
...
-import org.gradle.api.tasks.testing.logging.TestLogEvent
+@file:Suppress("UnstableApiUsage")
+
 import org.gradle.kotlin.dsl.kotlin
 import org.gradle.kotlin.dsl.withType
 
 plugins {
     id("kgraphql-base")
     kotlin("jvm")
+    `java-library`
```
Comment (mervyn-mccreight) [inline buildSrc/src/main/kotlin/kotlin-conventions.gradle.kts:9]: Why is this needed?

Comment (stuebingerb) [reply buildSrc/src/main/kotlin/kotlin-conventions.gradle.kts:9]: Because
> The plugin cannot be used without a JVM language plugin applied as it relies on several conventions of the `java` plugin.

https://docs.gradle.org/8.10.2/userguide/jvm_test_suite_plugin.html#sec:jvm_test_suite_usage

Tried, and it indeed fails.

Comment (stuebingerb) [reply buildSrc/src/main/kotlin/kotlin-conventions.gradle.kts:9]: No idea if `java` or `java-library` is more appropriate but we already use `java-library` so 🤷‍♂️

Comment (stuebingerb) [reply buildSrc/src/main/kotlin/kotlin-conventions.gradle.kts:9]: But we can now remove the java-library plugin in library-conventions.

**Review by mervyn-mccreight — APPROVED** [review]
